host_applications/linux/apps/raspicam/RaspiCamControl.c

```diff
@@ -1811,6 +1813,22 @@ void default_camera_control_callback(MMAL_PORT_T *port, MMAL_BUFFER_HEADER_T *bu
                         settings->awb_red_gain.num, settings->awb_red_gain.den,
                         settings->awb_blue_gain.num, settings->awb_blue_gain.den);
       }
+      case MMAL_PARAMETER_CAPTURE_STATUS:
+      {
+         wiringPiSetup();
+         pinMode(21, OUTPUT);
+         MMAL_PARAMETER_CAPTURE_STATUS_T *status = (MMAL_PARAMETER_CAPTURE_STATUS_T*)param;
+         if (status->status == MMAL_PARAM_CAPTURE_STATUS_CAPTURE_STARTED)
+         {
+            vcos_log_error("Started capture. Setting pin 21 (BCM pin 5) high...");
+            digitalWrite(21, HIGH);
+         }
+         else if (status->status == MMAL_PARAM_CAPTURE_STATUS_CAPTURE_ENDED)
+         {
+            vcos_log_error("Ended capture. Setting pin 21 (BCM pin 5) low...");
+            digitalWrite(21, LOW);
```

Comment:
I think we should still have a backup measure of turning the LED off in the python code after capture, just to be sure it ends up off.

Reply:
sure, since we still have that code :+1:

Reply:
yeah, we might want that somewhere where it will run after a Ctrl+C